var/spack/repos/builtin/packages/sim-model/package.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+from spack import *
+from contextlib import contextmanager
+import os, shutil
+
+
+class SimModel(Package):
+    """The abstract base package for simulation models.
+
+    Simulation models are groups of nmodl mechanisms. These packages are
+    deployed as neuron/coreneuron modules (dynamic loadable libraries)
+    which are loadable using load_dll() or linked into a "special"
+
+    Specific models packages can be added to spack by simply inheriting from
+    this class and defining basic attributes, e.g.:
+    ```
+    class ModelHippocampus(SimModel):
+        homepage = ""
+        git = "ssh://bbpcode.epfl.ch/sim/models/hippocampus"
+        version('develop', branch='master')
+    ```
+
+    Nevertheless, for them to become full neurodamus packages, they may inherit from
+    NeurodamusModel instead. See neurodamus-xxx packages for examples.
+
+    """
+    variant('coreneuron',  default=False, description="Enable CoreNEURON Support")
+    variant('profile',     default=False, description="Enable profiling using Tau")
+
+    depends_on('neuron~binary+mpi')
+    depends_on('coreneuron', when='+coreneuron')
+    depends_on('coreneuron+profile', when='+coreneuron+profile')
+    depends_on('neuron+profile', when='+profile')
+    depends_on('tau', when='+profile')
+
+    conflicts('^neuron~python', when='+coreneuron')
+
+    # -
+    phases = ('build', 'install')
+
+    # The name of the mechanism, which cen be overriden
+    mech_name = ''
+
+    # -
+    def build(self, spec, prefix):
+        link_flag = '-Wl,-rpath,' + prefix.lib
+        self._build_mods('mod', link_flag)
+
+    # -
+    def _build_mods(self, mods_location, link_flag='', include_flag='', corenrn_mods=None):
+        profile_flag = ' -DENABLE_TAU_PROFILER' if '+profile' in self.spec else ''
+        include_flag += profile_flag
+
+        if self.spec.satisfies('+coreneuron'):
+            mechlib = self.__build_mods_coreneuron(
+                corenrn_mods or mods_location, link_flag, include_flag)
+            # Link neuron special with this mechs lib
+            link_flag += ' ' + mechlib.ld_flags + self._get_lib_flags('coreneuron')
```

Comment:
better to put empty space between `mechlib.ld_flags + self._get_lib_flags` ?

Reply:
self._get_lib_flags already outputs with req. spaces